docs/make.jl

```diff
@@ -1,19 +1,87 @@
-    using Pkg
-    Pkg.activate(joinpath(@__DIR__, ".."))
-    Pkg.instantiate()
-    Pkg.activate(@__DIR__)
-    Pkg.instantiate()
-end
+if use_extra_workers > 0
+    Distributed.addprocs(use_extra_workers)
+
+    @everywhere begin
+        pushfirst!(LOAD_PATH, joinpath(@__DIR__, "..")) # add Oceananigans to environment stack
+
+        using Pkg
+        Pkg.activate(joinpath(@__DIR__, ".."))
+        Pkg.instantiate()
+        Pkg.activate(@__DIR__)
+        Pkg.instantiate()
+    end
+
+    @everywhere begin
+        using Documenter
+        using DocumenterCitations
+        using Literate
+
+        using CairoMakie # to avoid capturing precompilation output by Literate
+        CairoMakie.activate!(type = "svg")
+
+        using Oceananigans
+        using Oceananigans.Operators
+        using Oceananigans.Diagnostics
+        using Oceananigans.OutputWriters
+        using Oceananigans.TurbulenceClosures
+        using Oceananigans.TimeSteppers
+        using Oceananigans.AbstractOperations
+
+        using Oceananigans.BoundaryConditions: Flux, Value, Gradient, Open
+
+        bib_filepath = joinpath(dirname(@__FILE__), "oceananigans.bib")
+        bib = CitationBibliography(bib_filepath)
+
+        #####
+        ##### Generate examples
+        #####
+
+        const EXAMPLES_DIR = joinpath(@__DIR__, "..", "examples")
+        const OUTPUT_DIR   = joinpath(@__DIR__, "src/generated")
 
-@everywhere begin
+        examples = [
+            "One-dimensional diffusion"        => "one_dimensional_diffusion",
+            "Two-dimensional turbulence"       => "two_dimensional_turbulence",
+            "Internal wave"                    => "internal_wave",
+            "Convecting plankton"              => "convecting_plankton",
+            "Ocean wind mixing and convection" => "ocean_wind_mixing_and_convection",
+            "Langmuir turbulence"              => "langmuir_turbulence",
+            "Baroclinic adjustment"            => "baroclinic_adjustment",
+            "Kelvin-Helmholtz instability"     => "kelvin_helmholtz_instability",
+            "Shallow water Bickley jet"        => "shallow_water_Bickley_jet",
+            "Horizontal convection"            => "horizontal_convection",
+            "Tilted bottom boundary layer"     => "tilted_bottom_boundary_layer"
+        ]
+
+        example_scripts = [ filename * ".jl" for (title, filename) in examples ]
+
+        @info string("Executing the examples using ", Distributed.nprocs(), " processes")
+    end
+        
+    Distributed.pmap(1:length(example_scripts)) do n
+        example = example_scripts[n]
+        example_filepath = joinpath(EXAMPLES_DIR, example)
+        withenv("JULIA_DEBUG" => "Literate") do
+            Literate.markdown(example_filepath, OUTPUT_DIR;
+                            flavor = Literate.DocumenterFlavor(), execute = true)
+        end
+    end
+
+    Distributed.rmprocs()
+
+elseif use_extra_workers == 0
```

Comment:
Is there a way to avoid repeating this code?

Reply:
this is ONLY for benchmarking

I want to see how long the docs take with 0, 1, 2 extra workers

and then we can clean up this of course.

Reply:
(I could benchmark locally but it beats the purpose a bit... I want to see whether these changes have an effect on tartarus)